### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by patrickhammond**
Pulling in a handful of items listed here:
https://trello.com/c/ah4hvfwh/21-android-starter-project-updates

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -68,21 +66,21 @@ ext {
...
+    supportVersion = '24.2.1'
     libraryAndroidSupport = "com.android.support:support-v4:$supportVersion"
     libraryAndroidAppCompat = "com.android.support:appcompat-v7:$supportVersion"
     libraryAndroidRecyclerView = "com.android.support:recyclerview-v7:$supportVersion"
     libraryAndroidCardView = "com.android.support:cardview-v7:$supportVersion"
     libraryAndroidSupportAnnotations = "com.android.support:support-annotations:$supportVersion"
     libraryAndroidDesignSupport = "com.android.support:design:$supportVersion"
 
-    playServicesVersion = '8.4.0'
+    playServicesVersion = '9.4.0'
```
Comment (stuartsoft) [inline build.gradle]: FYI 9.6.1 is currently the newest version

File: app/src/main/java/com/mycompany/myapp/data/DataModule.java [hunk 2]
```diff
@@ -29,4 +39,46 @@ public OkHttpClient provideOkHttpClient(Cache cache) {
                 .cache(cache)
                 .build();
     }
+
+    @Singleton
+    @Api("github")
```
Comment (stuartsoft) [inline app/src/main/java/com/mycompany/myapp/data/DataModule.java]: Just curious, is this annotation coming from Retrofit? What's it used for?

File: app/src/main/java/com/mycompany/myapp/data/DataModule.java [hunk 3]
```diff
@@ -29,4 +39,46 @@ public OkHttpClient provideOkHttpClient(Cache cache) {
...
+                .client(client)
+                .baseUrl(baseUrl)
+                .addConverterFactory(converterFactory)
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .build();
+    }
+
+    @Singleton
+    @Provides
+    public GitHubApiService provideGitHubApiService(@Api("github") Retrofit retrofit) {
```
Comment (stuartsoft) [inline app/src/main/java/com/mycompany/myapp/data/DataModule.java]: `Just curious, is this annotation coming from Retrofit? What's it used for?`
answered my own question :P

File: build.gradle (hunk 1 above)
Comment (stuartsoft) [inline build.gradle]: Google has not yet released emulator images that support 9.6.1. We'll have to upgrade at a later date

File: app/src/dev/java/com/mycompany/myapp/app/VariantSettings.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        this.context = context;
+    }
+
+    public String getBaseUrl() {
+        String defaultBaseUrl = context.getString(R.string.default_base_url);
+        return getSettingsSharedPreferences().getString(PREF_BASE_URL, defaultBaseUrl);
+    }
+
+    public void setBaseUrl(String baseUrl) {
+        getSettingsSharedPreferences().edit().putString(PREF_BASE_URL, baseUrl).apply();
```
Comment (atoennis) [inline app/src/dev/java/com/mycompany/myapp/app/VariantSettings.java]: I like the variant settings setup here.  Definitely a useful option for any Android project.

File: app/src/main/java/com/mycompany/myapp/data/DataModule.java (hunk 3 above)
Comment (atoennis) [inline app/src/main/java/com/mycompany/myapp/data/DataModule.java]: Just to answer where it's coming from, it's defined under `com.getfinal.android.data.api` in main.  It's basically just saying this Retrofit instance is for github.  For instance, there could be another Retrofit instance with `@Api("bitbucket")` and that instance would be for bitbucket.

**Pull request merged**